### PR TITLE
feat: :sparkles: shell scripts to help list functions planned, made, needed

### PR DESCRIPTION
**Pull request by lwjohnst86**
## Description

This adds some shell scripts to make it easier to list functions we need. They don't have tests, they aren't super well documented so far (I don't know how to effectively document shell scripts), but they don't need to be too strict in structure since they only help us to develop and manage stuff.

Probably don't need to review it too much, mostly just to be aware of how to create and write these helpers.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -4,4 +4,4 @@
 # Install all scripts in `bin/` to `~/.local/bin/`
 install:
     mkdir -p ~/.local/bin
-    ln -s $(pwd)/bin/* ~/.local/bin
\ No newline at end of file
+    ln -sf $(pwd)/bin/* ~/.local/bin
```
Comment (signekb) [inline justfile:7]: What's the difference between `s` and `sf`?

**Review by signekb — COMMENTED** [review]
How do I test these? 😁

File: justfile (hunk 1 above)
Comment (lwjohnst86) [reply justfile:7]: sf is `soft link, force overwrite`. Otherwise it throws an error if it already exists and that's annoying.

**Comment by martonvago** [comment]
@lwjohnst86 Sorry, I'm a bit late on this, but just to say that these might not work completely on a mac with the default setup. E.g. it seems like there's no `-P` option for `grep`.

**Comment by martonvago** [comment]
Screenshot 2024-10-25 at 14 19 47

It is, but I get this ^^ when I run it. (Maybe I'm doing something else wrong though.)

**Pull request merged**